src/main/java/com/gravity9/mongocdc/MongoChangeStreamWorker.java

```diff
@@ -162,10 +162,19 @@ public void run() {
         } while (true);
     }
 
-    void register(ChangeStreamListener listener) {
-        log.info("Registering listener {} to worker on partition {} for collection {}", listener.getClass().getName(), partition, mongoConfig.getCollectionName());
-        listeners.add(listener);
-    }
+	void register(ChangeStreamListener listener) {
+		log.info("Registering listener {} to worker on partition {} for collection {}", listener.getClass().getName(), partition, mongoConfig.getCollectionName());
```

Comment:
I would replace here `listener.getClass().getName()` with `listener` or `listener.getName()` so client code can implement unique name for logging purposes.

Reply:
Right now - only toString is called. We have to think on how to implement it better